### ISSUE TITLE
alsa-scarlett-gui not starting unless root

### ISSUE THREAD

**Issue by kkflame** [geoffreybennett/alsa-scarlett-gui#141]
# `alsa-scarlett-gui` Issue Template

## Confirmation
- [x] I confirm that I have read the [FAQ](https://github.com/geoffreybennett/alsa-scarlett-gui/blob/master/FAQ.md).

## Issue Category
- [x] Bug Report

### Linux Distribution and Version
- Distribution: Endeavour OS
- Version: 2024.06.25

### Kernel Version
- Kernel version: 6.10.10-arch1-1

### Kernel Messages
[    2.482784] usb 1-2: New USB device found, idVendor=1235, idProduct=8213, bcdDevice= 6.45
[    2.482787] usb 1-2: New USB device strings: Mfr=1, Product=3, SerialNumber=2
[    2.482789] usb 1-2: Product: Scarlett 8i6 USB
[    2.482791] usb 1-2: Manufacturer: Focusrite
[    2.482792] usb 1-2: SerialNumber: F8R2Z0K9500FB3

[    5.904055] usb 1-2: Focusrite Scarlett Gen 3 Mixer Driver enabled (pid=0x8213); report any issues to https://github.com/geoffreybennett/scarlett-gen2/issues
[    5.931136] usb 1-2: Firmware version 1605

### Focusrite Interface Series and Model
Bus 001 Device 003: ID 1235:8213 Focusrite-Novation Scarlett 8i6 USB

### Audio System
- [x] PipeWire

## Issue Description
Please provide a detailed description of the issue or feature request, including steps to reproduce (if applicable), expected behavior, and actual behavior:

From a fresh EndeavourOS install I installed the alsa-scarlett-gui.

When I try to start the Gui normally I get the following error message:
Gdk-Message: 01:28:56.571: Error 71 (Protocol error) dispatching to Wayland display.

The Gui doesn't start.

If run the Gui as root I get the following message:
(alsa-scarlett-gui:7344): Gtk-WARNING **: 01:34:27.307: Unable to acquire session bus: Error spawning command line “dbus-launch --autolaunch=a7ac1fc277b24692ae95a1ce7f0510f5 --binary-syntax --close-stderr”: Child process exited with code 1

But the Gui starts and works fine.

Thought this might help.

Kind Regards.

**Comment by kkflame** [comment]
Almost forgot with the fresh install it gave me the additional error code:

(alsa-scarlett-gui:12717): Gtk-WARNING **: 02:22:20.250: Unknown key gtk-modules in /home/XXX/.config/gtk-4.0/settings.ini

If I comment out: #gtk-modules=colorreload-gtk-module:window-decorations-gtk-module in the settings.ini file this warning disappears.

Kind Regards.

**Comment by geoffreybennett** [comment]
I'll install Endeavour & see if I can reproduce this. What desktop environment are you using?

**Comment by kkflame** [comment]
Cool:

Operating System: EndeavourOS 
KDE Plasma Version: 6.1.5
KDE Frameworks Version: 6.6.0
Qt Version: 6.7.2
Kernel Version: 6.10.10-arch1-1 (64-bit)
Graphics Platform: Wayland
Processors: 16 × AMD Ryzen 7 5800X3D 8-Core Processor
Memory: 62,7 GiB of RAM
Graphics Processor: AMD Radeon RX 5700 XT

**Comment by geoffreybennett** [comment]
Sorry, I can't reproduce this. I installed Endeavour, clicked the button to do a system update since I did the offline version, ran:
- `sudo pacman -S gtk4`
- `git clone https://github.com/geoffreybennett/alsa-scarlett-gui`
- `cd alsa-scarlett-gui/src`
- `make -j4`
- `./alsa-scarlett-gui`

And it worked fine (same warning about gtk-modules that you saw).

I guess it's a bug or something not in my control :(. I found these which look related: https://bbs.archlinux.org/viewtopic.php?id=299488 https://gitlab.gnome.org/GNOME/gtk/-/issues/7015 — try `GSK_RENDERER=gl` or `ngl` or `GDK_BACKEND=x11 ./alsa-scarlett-gui` see if that helps?
